src/dex/balancer-v1/balancer-v1.ts

```diff
@@ -0,0 +1,729 @@
+  ): Promise<void> {
+    const _pools = pools === undefined ? this.eventPools.allPools : pools;
+
+    if (_pools.length === 0) throw Error('There are no pools.');
+
+    let addresses: string[][] = [];
+    let total = 0;
+
+    for (let i = 0; i < _pools.length; i++) {
+      let pool = _pools[i];
+
+      addresses.push([pool.id]);
+      total++;
+      pool.tokens.forEach(token => {
+        addresses[i].push(token.address);
+        total++;
+      });
+    }
+
+    let results = await this.eventPools.balancerMulticall.methods
+      .getPoolInfo(addresses, total)
+      .call({}, blockNumber);
+
+    let j = 0;
+    for (let i = 0; i < _pools.length; i++) {
+      _pools[i].tokens.forEach(token => {
+        token.balance = bignumberify(bmath.bnum(results[j]));
+        j++;
+      });
+    }
+  }
+
+  getPools(from: Token, to: Token, limit: number = 10): PoolState[] {
+    return this.eventPools.allPools
+      .filter(
+        p =>
+          p.tokens.some(
+            token => token.address.toLowerCase() === from.address.toLowerCase(),
+          ) &&
+          p.tokens.some(
+            token => token.address.toLowerCase() === to.address.toLowerCase(),
+          ),
+      )
+      .slice(0, limit);
+  }
+
+  async getPoolPrices(
+    amounts: bigint[],
+    side: SwapSide,
+    unitVolume: bigint,
+    exchangeProxy: Address,
+    pool: OldPool | null,
+  ): Promise<PoolPrices<BalancerV1Data> | null> {
+    if (!pool) return null;
+    try {
+      let unit = await this.eventPools.getPoolPrices(pool, side, unitVolume);
+
+      const _width = Math.floor((amounts.length - 1) / BALANCER_CHUNKS);
+      const _amounts = Array.from(Array(BALANCER_CHUNKS).keys()).map(
+        i => amounts[(i + 1) * _width],
+      );
+
+      const _prices = await Promise.all(
+        _amounts.map(a => this.eventPools.getPoolPrices(pool, side, a)),
+      );
+
+      const prices = interpolate(_amounts, _prices, amounts, side);
+
+      return {
+        prices,
+        unit,
+        data: {
+          pool: pool.id,
+          exchangeProxy,
+        },
+        poolAddresses: [pool.id],
+        exchange: this.dexKey,
+        poolIdentifier: BalancerV1.getIdentifier(this.dexKey, pool.id),
+        gasCost: BALANCER_SWAP_GAS_COST,
+      };
+    } catch (e) {
+      this.logger.error(`Error_getPoolPrices: ${pool.id}`, e);
+      return null;
+    }
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<BalancerV1Data>> {
+    try {
+      const _from = wrapETH(srcToken, this.network);
+      const _to = wrapETH(destToken, this.network);
+
+      const allPools = this.getPools(_from, _to);
+      const allowedPools = limitPools
+        ? allPools.filter(({ id }) =>
+            limitPools.includes(BalancerV1.getIdentifier(this.dexKey, id)),
+          )
+        : allPools;
```

Comment:
if you are limiting the pools to 10 pools at least you should take the top pools. When you call the getPoolIdentifers you should return the top n pools only. On the getPriceVolume if the limit pools is given simply check if those pools are valid (contain the src and dest tokens). If the limit pools is not given then get top n pools.

Reply:
I am not sure I follow. We can discuss this on standup :)